src/directives/bindInput.ts

```diff
@@ -227,20 +232,27 @@ class BindInputDirective extends AsyncDirective {
   private __getValueFromElement(element: Element): unknown {
     let value: unknown = undefined;
 
-    if (isInputElement(element)) {
-      if (element.type === 'checkbox') {
-        value = element.checked === true;
-      } else {
-        value = element.value;
+    if (isSelectElement(element)) {
+      switch (element.type.toLowerCase()) {
+        case 'select-multiple':
+          value = [...element.selectedOptions].map((opt) => opt.value);
+          break;
+        case 'select-one':
+        default:
+          value = element.value;
       }
-    } else if (isSelectElement(element)) {
-      if (element.multiple) {
-        value = [...element.selectedOptions].map((opt) => opt.value);
-      } else {
-        value = element.value;
+    } else if (isInputElement(element)) {
+      switch (element.type.toLowerCase()) {
+        case 'checkbox':
+          value = element.checked === true;
+          break;
+        case 'number':
+          value = Number(element.value);
```

Comment:
Why not use `valueAsNumber` instead?

Also, wouldn't you want to support `type=date` and the like too with `valueAsDate`?